internal/policy/targets_test.go

```diff
@@ -86,6 +86,53 @@ func TestUpdateDelegation(t *testing.T) {
 	}, targetsMetadata.Delegations.Roles[0])
 }
 
+func TestReorderDelegations(t *testing.T) {
+	targetsMetadata := InitializeTargetsMetadata()
+
+	key1, err := tuf.LoadKeyFromBytes(targets1PubKeyBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+	key2, err := tuf.LoadKeyFromBytes(targets2PubKeyBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	targetsMetadata, err = AddDelegation(targetsMetadata, "rule-1", []*tuf.Key{key1}, []string{"path1/"}, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	targetsMetadata, err = AddDelegation(targetsMetadata, "rule-2", []*tuf.Key{key2}, []string{"path2/"}, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Contains(t, targetsMetadata.Delegations.Keys, key1.KeyID)
+	assert.Contains(t, targetsMetadata.Delegations.Keys, key2.KeyID)
+	assert.Equal(t, key1, targetsMetadata.Delegations.Keys[key1.KeyID])
+	assert.Equal(t, key2, targetsMetadata.Delegations.Keys[key2.KeyID])
+	assert.Contains(t, targetsMetadata.Delegations.Roles, AllowRule())
+
+	ruleNames := []string{"rule-2", "rule-1"}
+	targetsMetadata, err = ReorderDelegations(targetsMetadata, ruleNames)
+	assert.Nil(t, err)
+
```

Comment:
let's set up a table of tests here:

```
	tests := map[string]struct {
		newOrder      []string
		expectedError error
	}{
		"reverse order": {...},
		"rule not specified in new order": {...},
		"rule repeated in new order": {...},
		"unknown rule in new order": {...},
		"allow rule in new order": {...},
	}
```

I think it's also easier for us to be sure we're testing every case, which I think isn't happening right now?

Reply:
See: https://github.com/gittuf/gittuf/blob/bb9ced2c12c8ec070e59c6f9d600ebc9a22197b1/internal/gitinterface/hash_test.go#L15-L61

Reply:
Will do!